File: CS/CurrentWindowServiceTest/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using System.Windows.Input;

namespace CurrentWindowServiceTest.ViewModel {
    public class MainViewModel : ViewModelBase {
        ICurrentWindowService CurrentWindowService { get { return GetService<ICurrentWindowService>(); } }
        public ICommand CloseWindowCommand { get; private set; }
        public MainViewModel() {
            CloseWindowCommand = new DelegateCommand(OnCloseWindowCommandExecute);
        }
        void OnCloseWindowCommandExecute() {
            CurrentWindowService.Close();
        }
    }
}

File: VB/CurrentWindowServiceTest/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System.Windows.Input

Namespace CurrentWindowServiceTest.ViewModel
    Public Class MainViewModel
        Inherits ViewModelBase

        Private ReadOnly Property CurrentWindowService() As ICurrentWindowService
            Get
                Return GetService(Of ICurrentWindowService)()
            End Get
        End Property
        Private privateCloseWindowCommand As ICommand
        Public Property CloseWindowCommand() As ICommand
            Get
                Return privateCloseWindowCommand
            End Get
            Private Set(ByVal value As ICommand)
                privateCloseWindowCommand = value
            End Set
        End Property
        Public Sub New()
            CloseWindowCommand = New DelegateCommand(AddressOf OnCloseWindowCommandExecute)
        End Sub
        Private Sub OnCloseWindowCommandExecute()
            CurrentWindowService.Close()
        End Sub
    End Class
End Namespace
